src/TreeTransform.jl

```diff
@@ -177,7 +179,6 @@ function bottom_up_rewrite(
         # Transform each node until it reaches a fixed-point.
         any_changes = false
```

Comment:
the docstring for `bottom_up_rewrite` indicates the constructors of the nodes are inspected to determine fields, but that no longer seems to be true - could you update it?